primitives/bridge/src/snowbridge_outbound_token_transfer.rs

```diff
@@ -0,0 +1,562 @@
+        }
+
+        // We only support a single asset at a time.
+        ensure!(reserve_assets.len() == 1, TooManyAssets);
+        let reserve_asset = reserve_assets.get(0).ok_or(AssetResolutionFailed)?;
+
+        // Fees are collected on Tanssi, up front and directly from the user, to cover the
+        // complete cost of the transfer. Any additional fees provided in the XCM program are
+        // refunded to the beneficiary. We only validate the fee here if its provided to make sure
+        // the XCM program is well formed. Another way to think about this from an XCM perspective
+        // would be that the user offered to pay X amount in fees, but we charge 0 of that X amount
+        // (no fee) and refund X to the user.
+        if let Some(fee_asset) = fee_asset {
+            // The fee asset must be the same as the reserve asset.
+            if fee_asset.id != reserve_asset.id || fee_asset.fun > reserve_asset.fun {
+                return Err(InvalidFeeAsset);
+            }
+        }
+
+        let (token, amount) = match reserve_asset {
+            Asset {
+                id: AssetId(inner_location),
+                fun: Fungible(amount),
+            } => match inner_location.unpack() {
+                (0, [AccountKey20 { network, key }]) if self.network_matches(network) => {
+                    Some((H160(*key), *amount))
+                }
+                _ => None,
+            },
+            _ => None,
+        }
+        .ok_or(AssetResolutionFailed)?;
+
+        // transfer amount must be greater than 0.
+        ensure!(amount > 0, ZeroAssetTransfer);
+
+        // Check if there is a SetTopic and skip over it if found.
+        let topic_id = match_expression!(self.next()?, SetTopic(id), id).ok_or(SetTopicExpected)?;
+
+        Ok((
+            Command::TransferNativeToken {
+                agent_id: self.agent_id,
+                token,
+                recipient,
+                amount,
+            },
+            *topic_id,
+        ))
+    }
+
+    fn next(&mut self) -> Result<&'a Instruction<Call>, XcmConverterError> {
+        self.iter
+            .next()
+            .ok_or(XcmConverterError::UnexpectedEndOfXcm)
+    }
+
+    fn peek(&mut self) -> Result<&&'a Instruction<Call>, XcmConverterError> {
+        self.iter
+            .peek()
+            .ok_or(XcmConverterError::UnexpectedEndOfXcm)
+    }
+
+    fn network_matches(&self, network: &Option<NetworkId>) -> bool {
+        if let Some(network) = network {
+            *network == self.ethereum_network
+        } else {
+            true
+        }
+    }
+
+    /// Convert the xcm for Polkadot-native token from the origin chain (container chain) into the Command
+    /// To match transfers of Polkadot-native tokens, we expect an input of the form:
+    /// # ReserveAssetDeposited
+    /// # ClearOrigin
+    /// # BuyExecution
+    /// # DepositAsset
+    /// # SetTopic
+    fn make_mint_foreign_token_command(
```

Comment:
I would for now perhaps error here. FIrst because it's untested (we have another way of transferring the tanssi tokens), so if we are going to open another path for this, I would like to be tested first. what do youy think?

Reply:
well in any case this is only goint to dancelight so I am fine with it. but let's create another issue please for adding tests for this route